pkg/apis/core/v1/defaults_test.go

```diff
@@ -43,6 +43,9 @@ import (
 // TestWorkloadDefaults detects changes to defaults within PodTemplateSpec.
 // Defaulting changes within this type can cause spurious rollouts of workloads on API server update.
 func TestWorkloadDefaults(t *testing.T) {
+	for _, feature := range utilfeature.DefaultFeatureGate.KnownFeatures() {
+		defer featuregatetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, feature, true)()
+	}
```

Comment:
Is it possible that **disabling** a feature gate could have an averse affect? I'm imagining something around a deprecated feature... If so, we may want 2 iterations of this test: one with all features enabled, and one with all disabled.

Reply:
I'm inclined to agree -- previously, we weren't setting feature gates, and I think we want coverage for both cases if possible.

Reply:
A run with features disabled requires a lot more extensive changes to distinguish locked features, since those fail if you try to disable them. I'm mostly trying to guard against default additions here.

Reply:
ha, as soon as I wrote that, I realized there's a trivial way to get the full registered feature spec

switched to that, added the requested positive and negative scenario, and dropped the changes to the feature gate interface